kusto/kql/query_parameters_test.go

```diff
@@ -0,0 +1,38 @@
+package kql
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestQueryParameters(t *testing.T) {
+	tests := []struct {
+		name     string
+		b        Builder
+		qp       *StatementQueryParameters
+		expected string
+	}{
+		{"Test empty", NewStatementBuilder(""), NewStatementQueryParameters(), "\n"},
+		{"Test add literal", NewStatementBuilder(""), NewStatementQueryParameters().AddLiteral("foo", "string", "bar"), "declare query_parameters(foo:string);\n"},
+		{"Test add identifiers", // test might fail at times due to the pseudo-random nature of map that will sometimes change the order of the declaration string.
+			NewStatementBuilder("").
+				AddDatabase("database").AddLiteral(".").
+				AddTable("table").AddLiteral(" | where ").
+				AddColumn("column").AddLiteral(" == ").
+				AddFunction("function").AddLiteral("() ;"),
+			NewStatementQueryParameters().
+				AddLiteral("database", "string", "foo_1").
+				AddLiteral("table", "string", "_bar").
+				AddLiteral("column", "string", "_baz").
+				AddLiteral("function", "string", "func_"),
+			"declare query_parameters(database:string, table:string, column:string, function:string);\ndatabase(\"database\").table | where column == function() ;"},
```

Comment:
WDYM? i agree addLiteral is a mistake, but wont AddString work?

Reply:
The parameters are typed, if you pass a parameter of type string with the value "a" to "table", it will be:
database(\"database\")."a"
Which will not work.

Try running the generated query.